src/traits.c

```diff
@@ -998,6 +998,7 @@ Expression *semanticTraits(TraitsExp *e, Scope *sc)
 #endif
                     }
 
+                    assert(strlen(sm->toChars()) > 0);
```

Comment:
Using `strlen` for the `Identifier` object is costly operation. Instead of that, you can use more lightweight comparison `sm->ident != Id::empty`.